x-pack/legacy/plugins/siem/server/utils/build_query/calculate_timeseries_interval.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import moment from 'moment';
+import { get } from 'lodash/fp';
+const d = moment.duration;
+
+const roundingRules = [
+  [d(500, 'ms'), d(100, 'ms')],
+  [d(5, 'second'), d(1, 'second')],
+  [d(7.5, 'second'), d(5, 'second')],
+  [d(15, 'second'), d(10, 'second')],
+  [d(45, 'second'), d(30, 'second')],
+  [d(3, 'minute'), d(1, 'minute')],
+  [d(9, 'minute'), d(5, 'minute')],
+  [d(20, 'minute'), d(10, 'minute')],
+  [d(45, 'minute'), d(30, 'minute')],
+  [d(2, 'hour'), d(1, 'hour')],
+  [d(6, 'hour'), d(3, 'hour')],
+  [d(24, 'hour'), d(12, 'hour')],
+  [d(1, 'week'), d(1, 'd')],
+  [d(3, 'week'), d(1, 'week')],
+  [d(1, 'year'), d(1, 'month')],
+  [Infinity, d(1, 'year')],
+];
+
```

Comment:
This looks very similar to things I see here:
```
x-pack/legacy/plugins/apm/server/lib/helpers/get_bucket_size/calculate_auto.js
```

And a few other places. No biggie if we have hard forks (as sometimes we do at the moment), I would just document where the hard fork was originally from so later if we see bugs or do compares we can understand the lineage (history) of the files.